lib/db_connection/ownership/manager.ex

```diff
@@ -98,6 +104,38 @@ defmodule DBConnection.Ownership.Manager do
   end
 
   @impl true
+  def handle_call(:get_connection_metrics, _from, %{pool: pool, owners: owners, log: log} = state) do
+    {:ok, pool_metrics} = DBConnection.ConnectionPool.get_connection_metrics(pool)
+
+    proxy_metrics =
+      owners
+      |> Enum.map(fn {_, {proxy, _, _}} ->
+        if Process.alive?(proxy) do
+          GenServer.call(proxy, :get_connection_metrics)
```

Comment:
Process.alive? Can always be a race because the process can die immediately after, so I would rather move the try/catch up here and catch any exit. :)

Reply:
I thought it would reduce the probability of race condition a little bit. 
I don't think it's possible to move `catch :exit, _` here because `DBConnection.ConnectionPool.get_connection_metrics` call may cause exit(e.g. due to timeout) too

Reply:
exactly, hence the suggestion of wrapping exactly that call.

Reply:
We would have to wrap both `ConnectionPool.get_connection_metrics` and `GenServer.call(proxy, :get_connection_metrics)` in that case. 
Wouldn't it be easier to catch all the exits in one place? 
Am I getting something wrong?

Reply:
We don't need to wrap the pool one. And wapping the second one means we can provide results even if one of the pools is failing.